ext/intl/formatter/formatter_format.c

```diff
@@ -103,6 +131,24 @@ PHP_FUNCTION( numfmt_format )
 			}
 			INTL_METHOD_CHECK_STATUS( nfo, "Number formatting failed" );
 			break;
+
+		case FORMAT_TYPE_DECIMAL:
+			if (!try_convert_to_string(number)) {
+				RETURN_THROWS();
+			}
+			// Convert string as a DecimalNumber, so we don't lose precision
+			formatted_len = unum_formatDecimal(FORMATTER_OBJECT(nfo), Z_STRVAL_P(number), Z_STRLEN_P(number), formatted, formatted_len, NULL, &INTL_DATA_ERROR_CODE(nfo));
+			if (INTL_DATA_ERROR_CODE(nfo) == U_BUFFER_OVERFLOW_ERROR) {
+				intl_error_reset(INTL_DATA_ERROR_P(nfo));
+				formatted = eumalloc(formatted_len);
+				unum_formatDecimal(FORMATTER_OBJECT(nfo), Z_STRVAL_P(number), Z_STRLEN_P(number), formatted, formatted_len, NULL, &INTL_DATA_ERROR_CODE(nfo));
+				if (U_FAILURE( INTL_DATA_ERROR_CODE(nfo) ) ) {
+					efree(formatted);
+				}
+			}
+			INTL_METHOD_CHECK_STATUS( nfo, "Number formatting failed" );
+			break;
+
 		case FORMAT_TYPE_CURRENCY:
 			if (getThis()) {
```

Comment:
Aside Nit: this could now be ``if (object)``

Reply:
I tried it, but it caused several test failures (seemingly `object` was still null when I wouldn’t have expected it to be).